mesonbuild/modules/__init__.py

```diff
@@ -167,6 +167,8 @@ def process_include_dirs(self, dirs: T.Iterable[T.Union[str, IncludeDirs]]) -> T
             else:
                 yield self._interpreter.build_incdir_object([d])
 
+    def add_language(self, lang: str, for_machine: MachineChoice, required: bool = True) -> None:
```

Comment:
Can you make the type for lang line up with the interpreter function? Too bad the argument order doesn't match too.

Reply:
The goal of this wrapper is to have a simplified API for modules instead of exposing the full interpreter API. I think for modules it makes more sense to add one language at a time and make it always required. I removed the required argument and forced True instead.

The reason add_langauges() is plurial in interpreter is because project() can add more than one.

Reply:
I agree with @xclaesse here. Moreover, the purpose of the module state methods is to seal modules away from the complexity of the interpreter... and allowing it to be optional is one form of complexity that feels unwarranted indeed.

Reply:
Works for me!